ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS.hs

```diff
@@ -305,28 +392,25 @@ localRootPeersProvider tracer timeout resolvConf rootPeersVar domains = do
             withResource' (TraceLocalRootFailure domain `contramap` tracer)
                           (1 :| [3, 6, 9, 12])
                           rr
-          reply <- lookupAWithTTL timeout resolvConf resolver daDomain
+          reply <- resolveDomain resolver domain advertisePeer
           case reply of
-            Left  err -> do
-              traceWith tracer (TraceLocalRootFailure domain (DNSError err))
-              go rrNext (ttlForDnsError err ttl)
-
+            Left err -> go rrNext (ttlForDnsError err ttl)
             Right results -> do
-              traceWith tracer (TraceLocalRootResult domain results)
               atomically $ do
-                rootPeers <- readTVar rootPeersVar
-                let resultsMap :: Map Socket.SockAddr PeerAdvertise
-                    resultsMap = Map.fromList [ ( Socket.SockAddrInet
-                                                    daPortNumber
-                                                    (IP.toHostAddress addr)
-                                                , advertisePeer)
-                                              | (addr, _ttl) <- results ]
-                    rootPeers' :: Map DomainAddress (Map Socket.SockAddr PeerAdvertise)
-                    rootPeers' = Map.insert domain resultsMap rootPeers
+                rootPeersGroups <- readTVar rootPeersGroupsVar
+                let rootPeersGroups' = IntMap.fromList $ zip [0,1..] rootPeersGroups
```

Comment:
Nice progress :)! Two more remarks:

* it seems as we could store the `IntMap` inside `rootPeersGroupsVar`
* it would also be cleaner if we'd use a `Group` newtype wrapper rather than a bare `Int`

Reply:
> it seems as we could store the IntMap inside rootPeersGroupsVar

Hmmm I guess we could but the type of rootPeersGroupsVar is meant to be the same as the diffusion interface type

> it would also be cleaner if we'd use a Group newtype wrapper rather than a bare Int

We can also do this but I'd like to avoid changing this now since that would impact config parameters parsing (the same for the suggestion above)

Reply:
In the long term it's better to merge clean things to `p2p-master` / `master`, otherwise we might end up with technical debt that we'll not clean for a long time. It shouldn't be too difficult to update the `Diffusion` interface.

Reply:
I understand that. In that case it might be better to do that here: https://github.com/input-output-hk/ouroboros-network/pull/3079 and then merge into this one and then, finally, merge into `p2p-master`. The scope of this PR wasn't to change anything in the Diffusion layer

Reply:
ok

Reply:
See: https://github.com/input-output-hk/ouroboros-network/pull/3079/commits/16ad70aa569026a07d3a76bbbd51e4fa48ba1a1a